week4/src/components/MainPage.jsx

```diff
@@ -0,0 +1,258 @@
+import { useState, useEffect, useRef } from "react";
+import styled from "styled-components";
+import axios from "axios";
+import StoreDataCard from "./StoreDataCard";
+
+function MainPage() {
+  const [beerStores, setBeerStores] = useState([]); //가게 정보 배열
+  const [isChecked, setIsChecked] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const location = useRef("");
+
+  useEffect(() => {
+    setIsLoading(true); //로딩중 초기화
+    setBeerStores([]); //가게 정보 담은 배열 초기화
+  }, []);
+
+  useEffect(() => {
+    if (isChecked) {
+      location.current.value = "";
+    }
+  }, [isChecked]);
+
+  function handleSearchButton() {
+    setIsLoading(true);
+
+    setTimeout(() => {
+      setIsLoading(false);
+    }, 1000);
+
+    if (!isChecked) {
+      //입력 지역 근처 검색
+      getDataNearByTown(location.current.value);
+    } else {
+      //사용자 현 위치 검색
+      getDataNearByUser();
+    }
+  }
```

Comment:
```suggestion
  function handleSearchButton() {
    setIsLoading(true);

    if (!isChecked) {
      //입력 지역 근처 검색
      getDataNearByTown(location.current.value);
    } else {
      //사용자 현 위치 검색
      getDataNearByUser();
    }
    
    setIsLoading(false);
  }
```

처음 `isLoading`값을 false로 초기화해주고 이렇게만 써줘도 원하는 대로 동작할 것 같아ㅏ!